kolibri/auth/models.py

```diff
@@ -0,0 +1,99 @@
+"""
+We have three main abstractions: Users, Collections, and Roles. Users represent people, like students in a school,
+teachers for a classroom, or volunteers setting up informal installations.
+
+Collections are hierarchical groups of users. Users belong to one or more Collections, and Collections can belong to
+other Collections. Collections are subdivided into several pre-defined levels.
+
+Roles belong to collections, and represent permissions. Users have one or more Roles. For instance, Classes (a type
+of Collection) have an associated Coach Role -- that Coach has permission to view related User data for Users in the
+Class.
+"""
+from __future__ import absolute_import, print_function, unicode_literals
+
+from django.contrib.auth.models import AbstractBaseUser
+from django.core import validators
+from django.db import models
+from django.utils.translation import ugettext_lazy as _
+from django.utils import timezone
+
+
+class BaseUser(AbstractBaseUser):
+    """
+    Our custom user type, derived from AbstractBaseUser as described in the Django docs.
+    Draws liberally from django.contrib.auth.AbstractUser, except we remove some fields we don't care about, like
+    email. Encapsulates both FacilityUsers and DeviceOwners, which are proxy models.
+
+    Do not use this class directly.
```

Comment:
Though perhaps "do not user directly" is a little stern, instead it should be "use with caution because the proxy models are meant to prevent you from making goofs."